tools/shared.py

```diff
@@ -2954,18 +2961,23 @@ def make_invoke(sig, named=True):
       args = ','.join(['a' + str(i) for i in range(1, len(legal_sig))])
       args = 'index' + (',' if args else '') + args
       ret = 'return ' if sig[0] != 'v' else ''
-      body = '%sModule["dynCall_%s"](%s);' % (ret, sig, args)
+      body = '%s%s(%s);' % (ret, JS.make_dynCall(sig), args)
```

Comment:
avoiding Module here (when not needed) is a separate optimization here? please add it the commit message too

Reply:
That's right, avoiding Module is separate from avoiding the `'longjmp'` case. Updated with comment